rbd/diff_iterate.go

```diff
@@ -118,15 +121,14 @@ func (image *Image) DiffIterate(config DiffIterateConfig) error {
 		C.uint64_t(config.Length),
 		C.uint8_t(config.IncludeParent),
 		C.uint8_t(config.WholeObject),
-		C.callDiffIterateCallback,
 		C.uintptr_t(cbIndex))
 
 	return getError(ret)
 }
 
 //export diffIterateCallback
 func diffIterateCallback(
-	offset C.uint64_t, length C.size_t, exists, index C.int) C.int {
+	offset C.uint64_t, length C.size_t, exists C.int, index C.uintptr_t) C.int {
```

Comment:
BTW: Today I found out, that - in case we don't go with a unsafe.Pointer here - you don't have to use `C.uintptr_t`, you can directly use `uintptr`. It doesn't seem to work for other types, but `C.uintptr_t` is automatically translated to `uintptr`. But I didn't check the docs, if we can rely on that behaviour.

Reply:
I took the recommendation to use unsafe.Pointer as an argument to the "static" callback function. Thus I  think I can resolve this.